File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    List<List<Integer>> output = new ArrayList<>();
    List<Integer> subset = new ArrayList<>();
    public List<List<Integer>> subsets(int[] nums) {

        dfs(nums, 0);

        return output;
    }

    private void dfs(int[] nums, int i) {
        if (i == nums.length) {
            output.add(new ArrayList<>(subset));
            return;
        }

        subset.add(nums[i]);
        dfs(nums, i+1);

        subset.remove(subset.size() - 1);
        dfs(nums, i+1);
    }
}

File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False] * n
        dp [-1] = True

        for i in range(n-2, -1, -1):
            for j in range(i+1, i+nums[i]+1):
                if j < n and dp[j]:
                    dp[i] = True
                if dp[i]:
                    break
        
        return dp[0]
